apps/studio/src/components/Datatable/Datatable.tsx

```diff
@@ -0,0 +1,167 @@
+  }
+}
+
+export const Datatable = <T extends object>({
+  instance,
+  isFetching,
+  pagination,
+  totalRowCount,
+  totalRowCountString,
+  tablePropOverrides,
+  emptyPlaceholder,
+  overflow = "auto",
+  ...tableProps
+}: DatatableProps<T>): JSX.Element => {
+  const { rows } = instance.getRowModel()
+  const styles = useMultiStyleConfig("Table", tableProps)
+
+  return (
+    <Flex flexDirection="column" layerStyle="shadow" pos="relative">
+      {isFetching && (
+        <>
+          <Flex
+            // white alpha to denote loading
+            bg="whiteAlpha.800"
+            bottom={0}
+            left={0}
+            p="1rem"
+            pos="absolute"
+            right={0}
+            top={0}
+            zIndex="1"
+          />
+          <Flex
+            bottom={0}
+            flex={1}
+            left={0}
+            pos="fixed"
+            right={0}
+            top={0}
+            w="100vw"
+            zIndex={2}
+          >
+            <Box m="auto">
+              <Spinner />
+            </Box>
+          </Flex>
+        </>
+      )}
+      <Box overflow={overflow} sx={styles.container}>
+        <Table sx={{ tableLayout: "fixed" }} {...tableProps} pos="relative">
+          <Thead>
+            {instance.getHeaderGroups().map((headerGroup) => (
+              <Tr
+                key={headerGroup.id}
+                // To toggle _groupHover styles to show divider when header is hovered.
+                data-group
+                borderBottomWidth="1px"
+              >
+                {headerGroup.headers.map((header) => (
+                  <Th
+                    key={header.id}
+                    pos="relative"
+                    px={0}
+                    style={{
+                      width:
+                        header.getSize() === Number.MAX_SAFE_INTEGER
+                          ? "auto"
+                          : header.getSize() === Number.MIN_SAFE_INTEGER
+                            ? "fit-content"
+                            : header.getSize(),
+                    }}
+                  >
+                    <Flex align="center">
+                      {flexRender(
+                        header.column.columnDef.header,
+                        header.getContext(),
+                      )}
+                    </Flex>
+                  </Th>
+                ))}
+              </Tr>
+            ))}
+          </Thead>
+          <Tbody>
+            {rows.length === 0 && emptyPlaceholder}
+            {rows.map((row) => {
+              return (
+                <Tr key={row.id} borderBottomWidth="1px">
+                  {row.getVisibleCells().map((cell) => {
+                    return (
+                      <Td
+                        key={cell.id}
+                        verticalAlign="center"
+                        {...tablePropOverrides?.[cell.column.id]?.td}
+                        {...tablePropOverrides?.[cell.id]?.td}
```

Comment:
not entirely sure i'm understanding this correctly here, but how would i know ahead of time what's the cell's specific id?

also, why allow an override of `td` props but not other components? i think if the use-case is narrow (only 1/2 places override `td`), we might as well just remove this prop since it's only for styling AHHA

Reply:
removing for now, can add back if we need to in the future. Current isomer's implementation does not need it yet.
(cell id provided by react-table. suggest reading react table docs on styling)